### ISSUE TITLE
Issue with Simple UDT Rule 3 in RFC 25 Simple UDT

### ISSUE THREAD

**Issue by phroi** [nervosnetwork/rfcs#395]
In [RFC 25 Simple UDT](https://github.com/nervosnetwork/rfcs/blob/master/rfcs/0025-simple-udt/0025-simple-udt.md#sudt-cell) **Simple UDT Rule 3** states:

> each SUDT must have unique type script, in other words, 2 SUDT cells using the same type script are considered to be the same SUDT.

On the other side @jordanmack, advising me on how to implement iCKB SUDT, wrote me this:

> In general, dapps will also be built to recognize tokens using the original SUDT implementation. Libraries too. If you look at [the code for PW-SDK, it includes the well-known SUDT script information](https://github.com/lay2dev/pw-core/blob/master/src/constants.ts#L29-L42).

Which lead me to believe the RFC is not in line with how this is implemented in tools supporting SUDT. In this regard what the **Simple UDT Rule 3** should really state is:

> each SUDT must have an unique pair of type script and owner lock, in other words, 2 SUDT cells using the same type script but different owner locks are considered to be the different SUDT.

Keep up the great work :muscle: 
Phroi

**Comment by phroi** [comment]
I see I see, now it makes sense, thanks!!

On the other side this should be explicit, especially since this is a RFC: this document is read both by navigated Nervos developers and people who are just learning L1, like me.

For example I have read many times this RFC and basically I misunderstood until now what was written there. Before Rule 3 didn't really make much sense, while now it does.

So in my opinion we should either improve the wording of that sentence or at least explain that here type script is comprehensive of its args, @contrun do you agree? If you disagree, what are your reasons?

**Comment by XuJiandong** [comment]
1. The Owner lock is optional. It is used while minting tokens.
2. The `script` includes field `args`. See [definition](https://docs.nervos.org/docs/reference/script/)

**Comment by phroi** [comment]
> 1. The Owner cell (which has owner lock script) is optional. It is used in transaction while minting tokens.

No problems on this, it's pretty clear from the RFC.

>     2. The `script` includes field `args`. See [definition](https://docs.nervos.org/docs/reference/script/)

That page indeed makes a good job of pointing out the difference between `script code` and `script`, basically that was my base mistake, thanks for pointing it out!! I skimmed thru the RFCs and I found no inconsistencies in this use, so all good on the RFC side.

Still my point keeps being the same: @XuJiandong what would be the harm in being a little bit more newbie friendly?

Could something as small as:

**Simple UDT Rule 3:** each SUDT must have unique type script [^1], in other words, 2 SUDT cells using the same type script are considered to be the same SUDT.

[^1]: As per definition a type script is comprised of three fields: `code_hash`, `hash_type` and`args`, see [script definition](https://docs.nervos.org/docs/reference/script/).
